atc/runtime/types.go

```diff
@@ -1,12 +1,87 @@
 package runtime
 
```

Comment:
Over time many of the runtime abstractions will fall in here, with k8s or worker package as the implementors. Every other package should only consume contents from runtime/types